jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/DiagnosticUtils.java

```diff
@@ -464,20 +464,20 @@ private static DiagnosticProperties doJFR(String diagnosticCommand) {
 		String fileName = parseStringParameter("filename", parameters, null);
 		IPC.logMessage("doJFR: filename = ", fileName);
 		boolean setFileName = (fileName != null) && !fileName.isEmpty();
-		if (setFileName) {
-			if (!VM.setJFRRecordingFileName(fileName)) {
-				return DiagnosticProperties.makeErrorProperties("setJFRRecordingFileName failed");
-			} else {
-				jfrRecordingFileName = fileName;
-			}
-		}
 
 		try {
 			if (command.equalsIgnoreCase(DIAGNOSTICS_JFR_START)) {
 				if (VM.isJFRRecordingStarted()) {
 					result = DiagnosticProperties.makeErrorProperties("One JFR recording is in progress [" + jfrRecordingFileName + "], only one recording is allowed at a time.");
 				} else {
 					VM.startJFR();
+					if (setFileName) {
+						if (!VM.setJFRRecordingFileName(fileName)) {
```

Comment:
this should be called before `openJFRFile` is called. So Ideally it is done before recording is started.

Reply:
Ignored the changing of the recording filename within the commands `JFR.dump` and `JFR.stop`.

Reply:
Fixed `setJFRRecordingFileName()` to allow the changing of `vm->jfrState.jfrFileName` before `initializaJFRWriter()`.